File: backend/mong-course/src/main/java/spring/edu/mongcourse/model/Course.java
package spring.edu.mongcourse.model;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Teacher
    private String name;
    private String description;
    private double price;
    private int credit;

    @ManyToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "categoryID")
    private Category category;

    public Course() {
    }

    public Course(Long id, String name, String description, double price, int credit, Category category) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.credit = credit;
        this.category = category;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getCredit() {
        return credit;
    }

    public void setCredit(int credit) {
        this.credit = credit;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

}

File: fontend/app-teacher/src/app/model/course.ts
export class Course {
  id: number;
  name: string;
  description: string;
  price: number;
  credit: number;
  categoryID: string;
  teacherID: string;

  constructor(
    id: number,
    name: string,
    description: string,
    price: number,
    credit: number,
    categoryID: string,
    teacherID: string
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.price = price;
    this.credit = credit;
    this.categoryID = categoryID;
    this.teacherID = teacherID;
  }
}
